src/corppa/poetry_detection/passim/build_passim_results.py

```diff
@@ -0,0 +1,217 @@
+                    break
+                # Skip unreferenced texts
+                ref_id = ref_record["id"]
+                if ref_id not in refs_to_pages[ref_corpus]:
+                    continue
+                for page_id in refs_to_pages[ref_corpus][ref_id]:
+                    # Update corresponding poem spans
+                    for span in page_records[page_id]["poem_spans"]:
+                        if (
+                            span["ref_corpus"] == ref_corpus
+                            and span["ref_id"] == ref_id
+                        ):
+                            start, end = span["ref_start"], span["ref_end"]
+                            span["ref_excerpt"] = ref_record["text"][start:end]
+
+
+def build_passim_output(
+    input_corpus,
+    passim_dir,
+    output_path,
+    disable_progress=False,
+    include_excerpts=False,
+    ref_corpora=None,
+):
+    # Get input ids
+    input_ids = {record["id"] for record in orjsonl.stream(input_corpus)}
+
+    # Initialize page-level annotation records
+    page_records = {}
+    for page_id in input_ids:
+        page_records[page_id] = {"page_id": page_id, "n_spans": 0, "poem_spans": []}
+
+    # Add passage-level matches to page-level records
+    for match in extract_passim_matches(
+        passim_dir, include_excerpts=include_excerpts, disable_progress=disable_progress
+    ):
+        page_id = match.pop("page_id")
+        page_records[page_id]["poem_spans"].append(match)
+        page_records[page_id]["n_spans"] += 1
+
+    # Optionally, add excerpts to page-level annotation records
+    if include_excerpts:
+        add_original_excerpts(
+            page_records,
+            input_corpus,
+            ref_corpora=ref_corpora,
+            disable_progress=disable_progress,
+        )
+
+    # Write output to file
+    orjsonl.save(output_path, page_records.values())
+
+
+def main():
+    """
+    Command-line access to build a JSONL file gathering the page-level passim results.
+    """
+    parser = argparse.ArgumentParser(description="Build page-level passim results.")
+
+    # Required arguments
+    parser.add_argument(
+        "input_corpus",
+        help="The input corpus used in the passim run (ppa corpus)",
+        type=pathlib.Path,
+    )
+    parser.add_argument(
+        "passim_output",
+        help="The top-level output directory for a passim run",
+        type=pathlib.Path,
+    )
+    parser.add_argument(
+        "output",
+        help="Filename for page-level passim output (JSON)",
+        type=pathlib.Path,
+    )
+    # Optional arguments
+    parser.add_argument(
+        "--progress",
+        help="Show progress",
+        action=argparse.BooleanOptionalAction,
+        default=True,
+    )
+    parser.add_argument(
+        "--include-excerpts",
+        help="Include text excerpts (original & aligned) for each match",
+        action="store_true",
```

Comment:
I vote for this being on by default. We could make it a boolean arg default true so you could do `--excerpts` / `--no-excerpts`

Reply:
I mean, we can just make it a required parameter; it's secondary because it serves not real purpose for evaluation. It's just nice for building the excerpt-level data.